assets/app/scripts/controllers/buildConfig.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:PodController
+ * @description
+ * # ProjectController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('BuildConfigController', function ($scope, $routeParams, DataService, project, BuildsService, $filter) {
+    $scope.buildConfig = null;
+    $scope.builds = {};
+    $scope.alerts = {};
+    $scope.renderOptions = $scope.renderOptions || {};    
+    $scope.renderOptions.hideFilterWidget = true;    
+    $scope.breadcrumbs = [
+      {
+        title: "Builds",
+        link: "project/" + $routeParams.project + "/browse/builds"
+      },
+      {
+        title: $routeParams.buildconfig
+      }
+    ];
+    $scope.emptyMessage = "Loading...";
+
+    var watches = [];
+    var objectWatches = [];
+
+    project.get($routeParams.project).then(function(resp) {
+      angular.extend($scope, {
+        project: resp[0],
+        projectPromise: resp[1].projectPromise
+      });
+      DataService.get("buildconfigs", $routeParams.buildconfig, $scope).then(
+        // success
+        function(buildConfig) {
+          $scope.buildConfig = buildConfig;
+
+          // If we found the item successfully, watch for changes on it
+          objectWatches.push(DataService.watchObject("buildconfigs", $routeParams.buildconfig, $scope, function(buildConfig, action) {
+            if (action === "DELETED") {
+              $scope.alerts["deleted"] = {
+                type: "warning",
+                message: "This build configuration has been deleted."
+              }; 
+            }
+            $scope.buildConfig = buildConfig;
+          }));           
+        },
+        // failure
+        function(e) {
+          $scope.alerts["load"] = {
+            type: "error",
+            message: "The build configuration details could not be loaded.",
+            details: "Reason: " + $filter('getErrorDetails')(e)
+          };
+        }
+      );
+
+      watches.push(DataService.watch("builds", $scope, function(builds, action, build) {
+        $scope.builds = {};
+        $scope.emptyMessage = "No builds to show";
+        // TODO we should send the ?labelSelector=buildconfig=<name> on the API request
+        // to only load the buildconfig's builds, but this requires some DataService changes
+        var allBuilds = builds.by("metadata.name");
+        angular.forEach(allBuilds, function(build, name) {
+          if (build.metadata.labels && build.metadata.labels.buildconfig === $routeParams.buildconfig) {
+            $scope.builds[name] = build;
+          }
+        });
+
+        var buildConfigName;
+        var buildName;
+        if (build) {
+          buildConfigName = build.metadata.labels.buildconfig;
```

Comment:
Is there a reason we check if `build.metadata.labels` is defined above but not here?